common/hsm_encryption.c

```diff
@@ -84,31 +86,42 @@ char *read_stdin_pass(char **reason)
 	char *passwd = NULL;
 	size_t passwd_size = 0;
 
-	/* Set a temporary term, same as current but with ECHO disabled. */
-	if (tcgetattr(fileno(stdin), &current_term) != 0) {
-		*reason = "Could not get current terminal options.";
-		return NULL;
+	if (isatty(fileno(stdin))) {
```

Comment:
Note that if this fails in `lightningd/options.c`, we'd fail. This is a bit unfortunate as we'd like to use it for `lightningd` as well i guess?

Reply:
yes, indeed the piped password still fails for `lightningd --encrypted_hsm`
It is because the temporary terminal is being created as a duplicate in https://github.com/ElementsProject/lightning/blob/03cfe0b468a550719918a7b56c0389e16c2cffb1/lightningd/options.c#L397 before calling the `read_stdin_pass`.
Should be able to clean that, testing now.

Reply:
Resolved with: https://github.com/ElementsProject/lightning/pull/4571/commits/8e86a57824107c11913490f84e78c9ea8c4f766b

tested ok with:
`$ (echo "test"; echo "test") | lightningd --encrypted-hsm`

and when called in the command line the password (echo) is still hidden
```
lightningd --encrypted-hsm
The hsm_secret is encrypted with a password. In order to decrypt it and start the node you must provide the password.
Enter hsm_secret password:
Confirm hsm_secret password:
```